axelrod/actions.py

```diff
@@ -15,3 +15,15 @@ def flip_action(action: Action) -> Action:
         return Actions.C
     else:
         raise ValueError("Encountered a invalid action.")
+
+
+def str_to_actions(actions: str) -> GeneratorType:
+    """Takes a string like 'cCdD' and returns a generator that yields the appropriate actions."""
+    for character in actions:
+        as_upper = character.upper()
+        if as_upper == 'C':
+            yield Actions.C
+        elif as_upper == 'D':
+            yield Actions.D
+        else:
+            raise ValueError('The characters of action_str may only be "C", "c", "D" and "d"')
```

Comment:
@marcharper mentioned this might be a thing to have and so i used the code he posted.  I found myself repeating code in cycler so I put this here.  it can easlily be pushed down to the two modules it's used in (cycler.py and test_cycler.py).  

it might be useful here if Actions ever gets refactored as a useful thing to go with it.  

it might be better as a list for fail-fast purposes.  this only raises an error once you iterate to offending character and not at instantiation (see tests).  if it returned a list, it would always fail immediately.